### PR TITLE
fix(playground): improve playground error handling

### LINKED ISSUE #5187: [playground] improve error handling

### PR DESCRIPTION
**Pull request by axiomofjoy**
https://github.com/user-attachments/assets/859af056-666f-4cb6-a92e-2eb055cb65b7

Improves error handling to yield an error type in the subscription so we can still return the final span at the end, even if an error occurs midway.

resolves #5187

### CODE REVIEW

File: tests/unit/server/api/test_subscriptions.py [hunk 1]
```diff
@@ -255,6 +264,126 @@ async def test_openai_text_response_emits_expected_payloads_and_records_expected
         ]
         assert not attributes
 
+    async def test_openai_emits_expected_payloads_and_records_expected_span_on_error(
```
Comment (Parker-Stafford) [inline tests/unit/server/api/test_subscriptions.py:267]: very nice

File: src/phoenix/server/api/subscriptions.py [hunk 2]
```diff
@@ -484,29 +491,35 @@ async def chat_completion(
...
+            events.append(
+                SpanException(
+                    timestamp=end_time,
+                    message=status_message,
+                    exception_type=type(error).__name__,
+                    exception_escaped=False,
+                    exception_stacktrace=format_exc(),
+                )
+            )
+            yield ChatCompletionSubscriptionError(message=status_message)
```
Comment (Parker-Stafford) [inline src/phoenix/server/api/subscriptions.py:510]: don't know if this matters as much on non SaaS, we could put a prefix on the error if it came from the llm provider, just to inform the user that the error did not happen in our system but happened in a third party api e.g., OpenAI API Error: status_message. Have to parse out where the error comes from which can be a pain though

File: app/src/pages/playground/PlaygroundOutput.tsx [hunk 3]
```diff
@@ -378,22 +395,25 @@ function PlaygroundOutputText(props: PlaygroundInstanceProps) {
     },
   });
 
-  if (!output && (toolCalls.length === 0 || instance.isRunning)) {
+  if (instance.isRunning) {
```
Comment (Parker-Stafford) [inline app/src/pages/playground/PlaygroundOutput.tsx:398]: nice fix

**Review by Parker-Stafford — APPROVED** [review]

File: src/phoenix/server/api/subscriptions.py (hunk 2 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/subscriptions.py:510]: I think this is a good idea. We don't currently have a display name in the backend for the providers, just a provider key (e.g., `openai` without the nice casing). Let's punt until we have.

**Pull request merged**